File: baekjoon/단계 별로 풀어보기/문자열/1316_그룹_단어_체커.py
import sys
input = sys.stdin.readline
IP = lambda: map(int, sys.stdin.readline().split())

def solve():
    result = 0
    
    for T in range(int(input())):
        word = input().rstrip()
        check = word[0]
        isit = True
        visited = set([word[0]])
        
        for letter in word:
            if check != letter:
                if letter in visited:
                    isit = False
                    break;
                
                visited.add(letter)
                check = letter
        
        result += isit
    
    print(result)

def other_solve():
    result = 0
    for T in range(int(input())):
        word = input().rstrip()
        result += (list(word) == sorted(word, key = word.find))
    print(result)
    
if "__main__" == __name__:
    #solve()
    other_solve()

File: baekjoon/단계 별로 풀어보기/문자열/1316_그룹_단어_체커.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main()
{
	cin.tie(NULL);
	cout.tie(NULL);
	ios_base::sync_with_stdio(false);
	
	int result = 0, T, isit = 1;
	string word;
	
	char check;
	
	cin >> T;
	
	while(T--)
	{
	    bool alpha[26] = {false};
	    cin >> word;
	    check = word[0];
	    isit = 1;
	    alpha[(int)word[0] - 97] = true;
	    
	    for (int j = 0; j < word.length(); j++)
	    {
	        if (check != word[j])
	        {
	            if (alpha[(int)word[j] - 97])
	            {
	                isit = 0;
	                break;
	            }
	            alpha[(int)word[j] - 97] = true;
	            check = word[j];
	        }
	    }
	    result += isit;
	}
	cout << result;
}